app/mailers/application_mailer.rb

```diff
@@ -7,4 +7,10 @@ class ApplicationMailer < ActionMailer::Base
   prepend IcsMultipartAttached
 
   append_view_path Rails.root.join("app/views/mailers")
+
+  # See https://www.bigbinary.com/blog/rails-5-2-allows-mailers-to-use-custom-active-job-class
+  class CustomMailerDeliveryJob < ActionMailer::DeliveryJob
+    discard_on ActiveJob::DeserializationError
+  end
+  self.delivery_job = CustomMailerDeliveryJob
```

Comment:
Avec cette approche on va rendre silencieux tous les échecs de déserialisation pour les mailers, et avec cette implémentation, c'est vraiment pas clair pourquoi on fait ça (alors qu'il me semble que c'est assez spécifique à la race condition sur la suppression des absences et plages d'ouvertures). Mais en même temps je me rends pas trop compte si c'est possible de faire autrement, est-ce qu'on peut ajouter un rescue spécifique pour les mailers concernés ? Changer le param passé au mailer pour prendre juste l'id et pas l'objet AR ? Ou autre chose ? Je serais curieux d'en discuter de vive voix.

Reply:
> Avec cette approche on va rendre silencieux tous les échecs de déserialisation pour les mailers, et avec cette implémentation, c'est vraiment pas clair pourquoi on fait ça (alors qu'il me semble que c'est assez spécifique à la race condition sur la suppression des absences et plages d'ouvertures). Mais en même temps je me rends pas trop compte si c'est possible de faire autrement, est-ce qu'on peut ajouter un rescue spécifique pour les mailers concernés ? Changer le param passé au mailer pour prendre juste l'id et pas l'objet AR ? Ou autre chose ? Je serais curieux d'en discuter de vive voix.

Je suis d'accord, et d'ailleurs je viens de voir qu'on avait eu aujourd’hui une remontée Sentry pour

    Error while trying to deserialize arguments: PG::ConnectionBad: PQconsumeInput() SSL SYSCALL error: EOF detected

et effectivement dans ce cas on veut retry !

Reply:
Je viens de modifier le code pour ne discard que quand on a une `ActiveRecord::RecordNotFound`. :man_dancing: